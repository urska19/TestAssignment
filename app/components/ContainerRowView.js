import React, {Component} from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import PropTypes from 'prop-types';

class ContainerRowView extends React.Component {

  static propTypes = {
    listRowContainerStyle: PropTypes.array,
    insideElement: PropTypes.element,
    onPress: PropTypes.func
  }

  static defaultProps = {
    listRowContainerStyle: [{
          height: 46,
          backgroundColor: 'white',
          borderRadius: 10,
          borderWidth: 1,
          marginVertical: 2,
          paddingHorizontal: 15
        }],
  }

  constructor(props) {
    super(props);
  }

  render() {
    return (
      <TouchableOpacity style={ this.props.listRowContainerStyle }
                        onPress={this.props.onPress}>
        {this.props.insideElement}
      </TouchableOpacity>
    )
  }
  
}

module.exports = ContainerRowView;
