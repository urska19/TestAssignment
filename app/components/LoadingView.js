import React, {Component} from 'react';
import {View, ActivityIndicator} from 'react-native';
import PropTypes from 'prop-types';

class LoadingView extends React.Component {

  static propTypes = {
    style: PropTypes.object,
    size: PropTypes.string,
    color: PropTypes.string
  }

  static defaultProps = {
    style: {alignItems: 'center', justifyContent: 'center'},
    size: 'large',
    color: 'yellow'
  }

  constructor(props) {
    super(props);
  }

  render() {
    return (
      <ActivityIndicator style={this.props.style} 
        size={this.props.size}
        color={this.props.color} />
    )
  }

}

module.exports = LoadingView;
