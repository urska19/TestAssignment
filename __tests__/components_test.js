import 'react-native';
import React from 'react';
import ContainerRowView from '../app/components/ContainerRowView';
import LoadingView from '../app/components/LoadingView';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <ContainerRowView />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  const tree = renderer.create(
    <LoadingView />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});