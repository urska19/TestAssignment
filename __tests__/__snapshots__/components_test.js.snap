// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 10,
      "borderWidth": 1,
      "height": 46,
      "marginVertical": 2,
      "opacity": 1,
      "paddingHorizontal": 15,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
/>
`;

exports[`renders correctly 2`] = `
<ActivityIndicator
  animating={true}
  color="yellow"
  hidesWhenStopped={true}
  size="large"
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
/>
`;
